app/src/main/java/com/realityexpander/tasky/presentation/splash_screen/SplashScreen.kt

```diff
@@ -0,0 +1,148 @@
+package com.realityexpander.tasky.presentation.splash_screen
+
+import androidx.compose.foundation.Image
+import androidx.compose.foundation.background
+import androidx.compose.foundation.layout.*
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Text
+import androidx.compose.runtime.*
+import androidx.compose.ui.Alignment
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.platform.LocalContext
+import androidx.compose.ui.res.colorResource
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.text.style.TextAlign
+import androidx.compose.ui.tooling.preview.Preview
+import androidx.compose.ui.unit.IntOffset
+import androidx.compose.ui.unit.dp
+import androidx.hilt.navigation.compose.hiltViewModel
+import com.ramcosta.composedestinations.annotation.Destination
+import com.ramcosta.composedestinations.annotation.RootNavGraph
+import com.ramcosta.composedestinations.navigation.DestinationsNavigator
+import com.realityexpander.tasky.R
+import com.realityexpander.tasky.common.settings.AppSettings
+import com.realityexpander.tasky.common.settings.setSettingsLoaded
+import com.realityexpander.tasky.dataStore
+import com.realityexpander.tasky.domain.AuthInfo
+import com.realityexpander.tasky.presentation.common.modifiers.largeHeight
+import com.realityexpander.tasky.presentation.destinations.AgendaScreenDestination
+import com.realityexpander.tasky.presentation.destinations.LoginScreenDestination
+import com.realityexpander.tasky.presentation.ui.theme.TaskyTheme
+import com.realityexpander.tasky.presentation.common.util.UiText
+import kotlinx.coroutines.delay
+import kotlinx.coroutines.launch
+
+@Composable
+@Destination
+@RootNavGraph(start = true)
+fun SplashScreen(
+    navigator: DestinationsNavigator,
+    viewModel: SplashViewModel = hiltViewModel()
+) {
+    val splashState by viewModel.splashState.collectAsState()
+    val scope = rememberCoroutineScope()
+    val context = LocalContext.current
+
+    val appSettings = context.dataStore.data.collectAsState( // .data is a Flow
+        initial = AppSettings()
+    ).value
```

Comment:
Here you can also use the `by` keyword to get rid of `.value`.

Also, you should be able to use `context.dataStore.data.drop(1).collectAsState(...)` to avoid the problem from below where you ignore the first emission.

Reply:
Good tip! I never know when I can use the `by` keyword.

Reply:
I get this warning when I try to use `.drop(1)` within a `@Composable`:

![image](https://user-images.githubusercontent.com/5157474/199136780-e2ba399a-854d-4056-9ee7-0eea44f8eb15.png)

Reply:
I found a different way to implement this that doesn't require a flow. It now works similar to how SharedPrefs operates.